stl/inc/vector

```diff
@@ -2880,6 +2880,423 @@ namespace pmr {
     using vector = _STD vector<_Ty, polymorphic_allocator<_Ty>>;
 } // namespace pmr
 #endif // _HAS_CXX17
+
+#if _HAS_IF_CONSTEXPR
+// VARIABLE TEMPLATE _Is_vb_iterator
+template <class _Alloc>
+_INLINE_VAR constexpr bool _Is_vb_iterator<_Vb_const_iterator<_Alloc>> = true;
+
+template <class _Alloc>
+_INLINE_VAR constexpr bool _Is_vb_iterator<_Vb_iterator<_Alloc>> = true;
+
+template <class _InIt, class _OutIt>
+_CONSTEXPR20 _OutIt _Copy_vbool(_InIt _First, _InIt _Last, _OutIt _Dest) {
+    // copy [_First, _Last) to [_Dest, ...)
+
+    const auto _DestEnd = _Dest + (_Last - _First);
+    _Vbase* _UFirst     = const_cast<_Vbase*>(_First._Myptr);
+    _Vbase* _UDest      = const_cast<_Vbase*>(_Dest._Myptr);
+    _Vbase* _ULast      = const_cast<_Vbase*>(_Last._Myptr);
+
+    // Fast path for less than _VBITS
+    // Always needed in case we only copy a range within one char
+    const bool _IsRightShift = _Dest._Myoff < _First._Myoff;
+    const auto _SourceShift  = _IsRightShift ? _First._Myoff - _Dest._Myoff : _Dest._Myoff - _First._Myoff;
+    if (_UFirst == _ULast) {
+        const bool _NeedsCarry = _DestEnd._Myptr != _Dest._Myptr;
+        const auto _SourceMask = (_Vbase(-1) << _First._Myoff) & (_Vbase(-1) >> _VBITS - _Last._Myoff);
+        const auto _SourceVal  = _IsRightShift ? (*_UFirst & _SourceMask) >> _SourceShift //
+                                              : (*_UFirst & _SourceMask) << _SourceShift;
+
+        const auto _DestMask = (_Vbase(-1) << _Dest._Myoff);
+        *_UDest              = (*_UDest & _DestMask) | _SourceVal;
+
+        if (!_NeedsCarry) {
+            return _DestEnd;
+        }
+
+        ++_UDest;
+        const auto _CarryMask = (_Vbase(-1) << _Last._Myoff - _DestEnd._Myoff) & (_Vbase(-1) >> _VBITS - _Last._Myoff);
+        const auto _CarryVal  = (*_UFirst & _CarryMask) >> _Last._Myoff - _DestEnd._Myoff;
+
+        const auto _DestCarryMask = _Vbase(-1) << _DestEnd._Myoff;
+        *_UDest                   = (*_UDest & _DestCarryMask) | _CarryVal;
+        return _DestEnd;
+    }
+
+    // _First and _Dest have matching char alignment, so use memmove
+    if (_First._Myoff & _Vbase(7) == _Dest._Myoff & _Vbase(7)) {

```

Comment:
```suggestion
    if ((_First._Myoff & _Vbase(7)) == (_Dest._Myoff & _Vbase(7))) {

```

Reply:
Thanks, I gave the names and pulled them out of the comparison